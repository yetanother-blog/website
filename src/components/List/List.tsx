import React from 'react';
import { Box } from '../Box/Box';
import { useTheme } from 'styled-components';

export const List: React.FC = ({ children }) => {
  const theme = useTheme();
  return (
    <Box as="ul" ml={theme.space.l} mb={theme.space.xl} p={0}>
      {children}
    </Box>
  );
};
