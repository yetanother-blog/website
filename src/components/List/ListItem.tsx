import React from 'react';
import { Box } from '../Box/Box';

export const ListItem: React.FC = ({ children }) => {
  return (
    <Box as="li" mb={0}>
      {children}
    </Box>
  );
};
