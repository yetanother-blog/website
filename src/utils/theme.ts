import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
