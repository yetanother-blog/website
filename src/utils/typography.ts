import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
