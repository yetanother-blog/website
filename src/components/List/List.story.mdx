import { Story, Meta, Preview, Props } from '@storybook/addon-docs/blocks';
import { List } from './List';
import { ListItem } from './ListItem';

<Meta title="Components|List" />

# List

Simple List Component for now

<Preview>
  <Story name="default">
    <List>
      <ListItem>Hello One</ListItem>
      <ListItem>Hello Two</ListItem>
      <ListItem>Hello Hundred</ListItem>
    </List>
  </Story>
</Preview>
