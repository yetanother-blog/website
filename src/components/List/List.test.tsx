import React from 'react';
import { render } from '@testing-library/react';
import { List } from './List';
import { ListItem } from './ListItem';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('List component', () => {
  it('should render correctly w/ list item', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <List>
          <ListItem>Hello One</ListItem>
          <ListItem>Hello Two</ListItem>
          <ListItem>Hello Hundred</ListItem>
        </List>
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
