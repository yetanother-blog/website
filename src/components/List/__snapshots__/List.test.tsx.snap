// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render correctly w/ list item 1`] = `
<div>
  <ul
    class="sc-AxjAm hkumP"
  >
    <li
      class="sc-AxjAm cPBeYE"
    >
      Hello One
    </li>
    <li
      class="sc-AxjAm cPBeYE"
    >
      Hello Two
    </li>
    <li
      class="sc-AxjAm cPBeYE"
    >
      Hello Hundred
    </li>
  </ul>
</div>
`;
